### DISCUSSION TITLE
React Native support

### DISCUSSION THREAD

**Discussion by corysimmons** [moalamri/vscode-inline-fold#107] [Ideas]
I'm not sure if it'd be an easy thing to upgrade, but it doesn't seem to work with React Native.

The only difference I can think of between normal React and React Native (so far as this plugin goes) is `className` needs to be added as a custom prop in React Native. I'm guessing this plugin iterates through a list of HTML DOM elements, whereas in React Native World those could be things like `View` (instead of `div`), `<Text>` (instead of `p` or `h1`), etc.

My use case is someone figured out how to bring Tailwind to React Native and lots of people are using this (and RN) now: https://www.nativewind.dev/

**Comment by moalamri** [comment]
I haven't used React Native, but this extension is (in purpose) not language or syntax specific. It doesn't walk through your code blocks, it simply uses RegExp against text. It should definitely work with the syntax I've seen here https://www.nativewind.dev/. If it doesn't work out of the box, then it must be the opened file (React Native file) not being added to supported languages list (which can be added manually in the extension's settings) under `"inlineFold.supportedLanguages": ["javascriptreact", "typescriptreact" .... "react-native-language-id"]`
You can find the language id in the bottom status bar at the right:
<img width="125" alt="image" src="https://user-images.githubusercontent.com/964077/221080054-78f0dab4-f8ce-46d1-8c28-de6297d07d4d.png">
See what is the language name in the status bar and add it to the extension's languages in lower case.
